dotcom-rendering/src/lib/useAdBlockAsk.ts

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from 'react';
+import { useAB } from './useAB';
+
+const useIsInAdBlockAskVariant = (): boolean => {
+	const abTestAPI = useAB()?.api;
+	const isInVariant = !!abTestAPI?.isUserInVariant('AdBlockAsk', 'variant');
+	return isInVariant;
+};
+
+const detectByRequests = async () => {
+	/**
+	 * Make a HEAD request to a URL that is typically blocked by ad-blockers
+	 */
+	const requestDoubleclick = async () => {
+		try {
+			const response = await fetch('https://www3.doubleclick.net', {
+				method: 'HEAD',
+				mode: 'no-cors',
+				cache: 'no-store',
+				signal: AbortSignal.timeout(1000),
+			});
+
+			// A redirect is another clue we may be being ad-blocked
+			if (response.redirected) {
+				return false;
+			}
+
+			return true;
+		} catch (err) {
+			return false;
+		}
+	};
+
+	/**
+	 * Make a HEAD request to a URL that should succeed, even when using an
+	 * ad-blocker
+	 *
+	 * We set this request with a much smaller timeout than the one we expect to
+	 * be ad-blocked. This should reduce the chance that request fails and this
+	 * one succeeds due to poor network connectivity
+	 */
+	const requestGuardian = async () => {
```

Comment:
Would it make sense to receive the timeout as a parameter, even if using a default value of `250`?